index.bs

```diff
@@ -314,6 +354,49 @@ The <dfn method for=FileSystemFileHandle>createWritable(|options|)</dfn> method,
 
 </div>
 
+### The {{FileSystemFileHandle/createSyncAccessHandle()}} method ### {#api-filesystemfilehandle-createsyncaccesshandle}
+
+<div class="note domintro">
+  : |handle| = await |fileHandle| . {{FileSystemFileHandle/createSyncAccessHandle()|createSyncAccessHandle}}()
+  :: Returns a {{FileSystemSyncAccessHandle}} that can be used to read/write from/to the file.
+     Changes made through |handle| might be immediately reflected in the file represented by |fileHandle|.
+     To ensure the changes are reflected in this file, the handle must be flushed or closed.
+
+     Creating a {{FileSystemSyncAccessHandle}} [=file entry/lock/take|takes an exclusive lock=] on the
+     [=FileSystemHandle/entry=] associated with |fileHandle|. This prevents the creation of
+     further {{FileSystemSyncAccessHandle}}s or {{FileSystemWritableFileStream}}s
+     for the entry, until the access handle is closed.
+
+     The returned {{FileSystemSyncAccessHandle}} offers synchronous {{FileSystemSyncAccessHandle/read()}} and
+     {{FileSystemSyncAccessHandle/write()}} methods. This allows for higher performance for critical methods on
+     contexts where asynchronous operations come with high overhead, e.g., WebAssembly.
+
+     For the time being, this method will only succeed when the |fileHandle| belongs to the
+     [=origin private file system=].
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemFileHandle>createSyncAccessHandle()</dfn> method, when invoked, must run these steps:
+
+1. Let |result| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+  1. Let |access| be the result of running [=this=]'s [=FileSystemHandle/entry=]'s
+     [=entry/request access=] given "`readwrite`".
+     If that throws an exception, [=reject=] |result| with that exception and abort.
+  1. If |access| is not "{{PermissionState/granted}}",
+     reject |result| with a {{NotAllowedError}} and abort.
+  1. Let |entry| be [=this=]'s [=FileSystemHandle/entry=].
+  1. If |entry| does not represent an [=/entry=] in an [=origin private file system=],
```

Comment:
Does using "NotAllowedError" actually solve the problem? The first guard reuses the exception from "request access" (if any) and it seems like that could in theory throw arbitrary exceptions.

Reply:
> The implementation in Chrome currently returns `InvalidStateError` if you try to create an AccessHandle ... on a locked file

I would prefer to see `NoModificationAllowedError` used when the file is inaccessible (which is what this PR suggests), either due to locking by the API or because the file is not readable/writable on the underlying file system (which should only be relevant outside of the OPFS). I'll update the web platform tests accordingly.

Reply:
I think one important thing to consider here is if callers are actually expected to branch on the different exception types. If not, I don't think there is much value in trying to distinguish the cases and it might even make certain optimizations harder.

Reply:
SGTM! That matches our implementation :)